File: src/classes/Candidate.java
package src.classes;
import java.util.Date;
import java.util.LinkedList;
import src.Election;


public class Candidate {
	// attributes
	private int number = 0;
	private int politicalPartyNumber = 0;
	private String name = "";
	private String ballotBoxAlias = "";
	private int votes = 0;
	private String votesDestiny = "";
	private String situation = "";
	private char gender = ' ';
	private Date birthDate = null;
	private int position = 0;


	// constructor method
	public Candidate(String ballotBoxAlias) {
		this.ballotBoxAlias = ballotBoxAlias;
	}


	// methods
	public void displayCandidateInfo() {
		System.out.println("Candidate alias: " + ballotBoxAlias);
		System.out.println("Candidate number: " + number);
		System.out.println("Candidate name: " + name);
		System.out.println("Candidate votes: " + votes);
		System.out.println("Candidate votes destiny: " + votesDestiny);
		System.out.println("Candidate situation: " + situation);
		System.out.println("Candidate political party number: " + politicalPartyNumber);
		System.out.println("Candidate gender: " + gender);
		Election info = new Election();
		System.out.println("Candidate birthdate: " + info.parseDateToString(birthDate));
		System.out.println("Candidate position: " + position);
		System.out.println();
	}

	public int getNumber() {
		return number;
	}
	public void setNumber(int number) {
		this.number = number;
	}

	public int getPoliticalPartyNumber() {
		return politicalPartyNumber;
	}
	public void setPoliticalPartyNumber(int politicalPartyNumber) {
		this.politicalPartyNumber = politicalPartyNumber;
	}
	public String getPoliticalPartyNameByNumber(LinkedList<Party> parties) {
		for (Party party : parties) {
			if (party.getNumber() == politicalPartyNumber) {
				return party.getName();
			}
		}
		return "";
	}
	public String getPoliticalPartyAliasByNumber(LinkedList<Party> parties) {
		for (Party party : parties) {
			if (party.getNumber() == politicalPartyNumber) {
				return party.getAlias();
			}
		}
		return "";
	}

	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}

	public String getBallotBoxAlias() {
		return ballotBoxAlias;
	}
	public void setBallotBoxAlias(String ballotBoxAlias) {
		this.ballotBoxAlias = ballotBoxAlias;
	}

	public int getVotes() {
		return votes;
	}
	public void setVotes(int votes) {
		this.votes = votes;
	}
	public void addVotes(int votes) {
		this.votes += votes;
	}

	public String getVotesDestiny() {
		return votesDestiny;
	}
	public void setVotesDestiny(String votesDestiny) {
		this.votesDestiny = votesDestiny;
	}

	public String getSituation() {
		return situation;
	}
	public void setSituation(String situation) {
		this.situation = situation;
	}

	public char getGender() {
		return gender;
	}
	public void setGender(char gender) {
		this.gender = gender;
	}

	public Date getBirthDate() {
		return birthDate;
	}
	public void setBirthDate(Date birthDate) {
		this.birthDate = birthDate;
	}

	public int getPosition() {
		return position;
	}
	public void setPosition(int position) {
		this.position = position;
	}
}

File: src/candidate.cpp
#include "../include/candidate.h"


Candidate::Candidate(const string &ballotBoxAlias) {
	this->ballotBoxAlias = ballotBoxAlias;
}


void Candidate::displayCandidateInfo() {
	cout << "Candidate alias: " << ballotBoxAlias << endl;
	cout << "Candidate number: " << number << endl;
	cout << "Candidate name: " << name << endl;
	cout << "Candidate votes: " << votes << endl;
	cout << "Candidate votes destiny: " << votesDestiny << endl;
	cout << "Candidate situation: " << situation << endl;
	cout << "Candidate political party number: " << politicalPartyNumber << endl;
	cout << "Candidate gender: " << gender << endl;
	Utils util = Utils();
	cout << "Candidate birthdate: " << util.parseTimeToString(birthDate) << endl;
	cout << "Candidate position: " << position << endl;
	cout << "\n";
}

const int Candidate::getNumber() {
	return number;
}
void Candidate::setNumber(const int number) {
	this->number = number;
}

const int Candidate::getPoliticalPartyNumber() {
	return politicalPartyNumber;
}
void Candidate::setPoliticalPartyNumber(const int politicalPartyNumber) {
	this->politicalPartyNumber = politicalPartyNumber;
}

const string& Candidate::getName() {
	return name;
}
void Candidate::setName(const string &name) {
	this->name = name;
}

const string& Candidate::getBallotBoxAlias() {
	return ballotBoxAlias;
}
void Candidate::setBallotBoxAlias(const string &ballotBoxAlias) {
	this->ballotBoxAlias = ballotBoxAlias;
}

const int Candidate::getVotes() {
	return votes;
}
void Candidate::setVotes(const int votes) {
	this->votes = votes;
}
void Candidate::addVotes(const int votes) {
	this->votes += votes;
}

const string& Candidate::getVotesDestiny() {
	return votesDestiny;
}
void Candidate::setVotesDestiny(const string &votesDestiny) {
	this->votesDestiny = votesDestiny;
}

const string& Candidate::getSituation() {
	return situation;
}
void Candidate::setSituation(const string &situation) {
	this->situation = situation;
}

const char Candidate::getGender() {
	return gender;
}
void Candidate::setGender(char gender) {
	this->gender = gender;
}

const time_t Candidate::getBirthDate() {
	return birthDate;
}
void Candidate::setBirthDate(time_t birthDate) {
	this->birthDate = birthDate;
}

const int Candidate::getPosition() {
	return position;
}
void Candidate::setPosition(const int position) {
	this->position = position;
}
